### DISCUSSION TITLE
Icaros 3.3.2

### DISCUSSION THREAD

**Discussion by Xanashi** [Xanashi/Icaros#82] [Releases]
Good Afternoon Everyone,

We're closing in on the end of the year and I somehow managed to get out one more 2023 release of Icaros.
This time I'm skipping the Beta phase, as I don't believe any of the changes will cause any regressions.
Check the changelog for all the details.

I hope you've all had a wonderful fall, and as always thank you all so much for the donations, 
troubleshooting and help with the translations! ❤️ 

I hope you all enjoy the new release!

<br/>

[![Icaros releases](https://img.shields.io/github/downloads/xanashi/icaros/v3.3.2/total?color=%232B6870&label=DOWNLOADS&style=for-the-badge)](https://github.com/Xanashi/Icaros/releases/tag/v3.3.2)

_______________________________________________________________________________________________________

### Icaros v.3.3.2 Final

- Added partial support for HEIF/HEIC thumbnailing 
- Added support for parsing iTunes tags in MOV/MP4 files (Make, Model and GPS)
- Added support for parsing the Keywords, Director and Producer tags in MKV
- Fixed not being able to set the Icaros Cache location on a FAT, FAT32 or ExFAT drive
- Fixed failing to read properties of certain MOV/MP4 files
- Fixed wrong duration and bitrate showing for OGG-Opus/Opus files
- Fixed long path files not working when debugging with 'Properties via Explorer'
- Fixed 'Max Cache Size' or 'Min Free Space' not working in the Indexer immediately after setting 
- Added Russian installer localization (Thank you Долматов and Ligre! ^__^)
- Updated French localization (Thank you rico.sos! ^__^)
- Replaced Flattr with Ko-fi donation link
- Updated LibDav1d to 1.3.0
- Updated GCC 
- Updated FFmpeg

<hr /><em>This discussion was created from the release <a href='https://github.com/Xanashi/Icaros/releases/tag/v3.3.2'>Icaros 3.3.2</a>.</em>

**Comment by iDolmatov** [comment]


**Reply by Xanashi** [reply]
You're absolutely right. Copy-paste oversight!
I'll fix it right away. Sorry for the misprint.

**Comment by chunme** [comment]
win10 LTSC2021 系统下，加载了视频缩略图，但是打开文件资源管理器不起作用。还是会加载缩略图很慢甚至最后卡住。

**Reply by Xanashi** [reply]
I don't have a LTSC2021 win10 build readily available for testing. 
Do you get any errors during thumbnailing, or is it just stuck?
Can you provide any more info, f.x. about the files you're trying to thumbnail?

**Reply by chunme** [reply]
就是正常的电影视频mkv\mp4文件，win10的文件管理器加载缩略图非常慢，开始以为是机械硬盘的缘故，把文件挪 P41 固态硬盘下也是如此。试了款 Media Preview软件，创建缓存卡在4K视频处。在吾爱破解论坛看到 Icaros；缓存可以正常创建功能也启用了，dll也注册了，软件开启状态和不开启状态打开文件管理器的电影文件夹，还是卡在加载缩略图。安装版与便携版都试过了。感觉无法调用是我哪里设置不对么？

**Reply by Xanashi** [reply]
I don't believe there is anything wrong with your settings, unless LTSC2021 Win10 has a different shell setup or different permissions than the normal Win10 builds. I can't be sure without testing. 
With that said, it is probably more likely that the slowdown is caused by one or more files, that Icaros is having a hard time processing,
or even crashes on. 
You could try to open IcarosConfig, and go to the Tools page, the select the Debug tab.
Drop you files into this window, and try running the "Thumbnailing via Icaros" test on the files. 
See if any errors occur.

**Reply by showmeren** [reply]
I too have been struggling with the same problem for a while.

Windows Explorer crashes,
Huge window lag,
Access to the video file itself was impossible.
Did you catch a virus? It makes me think.

RAM usage and disk usage were close to 100%.


I checked all versions of Icaros one by one, but this problem is not a problem with the Icaros program but a Windows 10 bug.

The solution is to change the ts video file extension.
You can change it to mp4, mkv, etc.

If the producer wants to test the same environment,
In an environment with 10 to 30 large TS video files of 5GB to 15GB.
If you implement it, you can reproduce this problem exactly.

Thank you, Icaros creator, for making a great program.

I hope that anyone with the same symptoms as me can resolve the Windows 10 ts file bug with my answer.


I will attach a link

https://answers.microsoft.com/en-us/windows/forum/all/windows-10-hanging-when-selecting-large-ts/dfdc431d-6590-49dc-91f0-7c604da10fa3
https://appuals.com/80-ram-and-disk-usage-by-windows-explorer-when-loading-folders-containing-ts-files/
https://answers.microsoft.com/en-us/windows/forum/windows_10-files-winpc/windows-10-file-explorer-always-scans-a-ts-file-on/ddd6d412-ba2a-48c2-8fa2-64a9c07d502c
https://answers.microsoft.com/en-us/insider/forum/all/100-disk-95-memory-system-unusable-windows-10-64/39ad23b2-b81b-4fcf-933f-f1edfab11ac2
https://www.reddit.com/r/techsupport/comments/gdhcsx/problem_with_large_ts_video_files_crashing/?rdt=39011
https://superuser.com/questions/1304114/windows-explorer-hangs-creating-thumbnails-for-typescript-files
https://answers.microsoft.com/en-us/windows/forum/all/windows-10-file-explorer-always-scans-a-ts-file-on/ddd6d412-ba2a-48c2-8fa2-64a9c07d502c
https://stackoverflow.com/questions/42475126/visual-studio-and-or-windows-explorer-crash-with-typescript-files
https://github.com/microsoft/TypeScript/issues/47611
https://superuser.com/questions/1304114/windows-explorer-hangs-creating-thumbnails-for-typescript-files/1733384#1733384
https://wizcorp.tumblr.com/post/176015676507/fix-inside-why-windows-explorer-goes-funky-in
https://answers.microsoft.com/en-us/windows/forum/all/mpeg-ts-file-in-preview-pane-of-file-explorer/d471aff2-c6e3-467c-a764-5efca439d8fb

**Reply by chunme** [reply]
非常感谢你的回答，我的电影文件夹里面没有TS媒体文件。都是 mp4 和 mkv 后缀的。
即使把 Icaros 生成缩略图的文件夹改成系统默认文件夹C:\Users\CC\AppData\Local\Microsoft\Windows\Explorer
也还是不能被系统识别并调用。

最后没办法只能用 MediaPreview 
虽然这个软件很老，不能排除C盘，生成缩略图也很慢很慢，4K视频还不能生成。
但是用 MediaPreview 生成缩略图显示正常，打开文件夹也变快了。
再安装 K-Lite Codec Pack 生成4K视频缩略图。彻底解决了我的问题。

（我甚至将 K-Lite Codec Pack 安装目录下的 K-Lite Codec Pack\Icaros\64-bit\IcarosPropertyHandler.dll 放到 Icaros\64-bit 目录下也不行）

**Reply by clsid2** [reply]
The Icaros uses by K-Lite is identical to standalone Icaros. But it does not enable the Icaros cache option by default. Meaning it only uses the standard Windows thumbnail cache. So likely your problem is an issue with access permissions of the cache folder.

**Reply by chunme** [reply]
具体我也搞不明白，Icaros创建的是 .icdb 后缀的缓存文件。MediaPreview创建的 .db 后缀就可以

**Reply by Xanashi** [reply]
The 'Icaros Cache' creates is a "backup" cache. It isn't needed to generate thumbnails in Explorer with Icaros, it just helps with speeding up thumbnailing in case the Windows thumbnail cache is lost. 

The Icaros thumbnail provider also simply generates the .db files (via Explorer) in the C:\Users\CC\AppData\Local\Microsoft\Windows\Explorer folder.
You can try to disable the 'Icaros Cache' and see that it still works, and modifies the .db files.

Now as for your problem, I'm not sure if Google Translate really does your questions justice. I may not fully understand what the issue is.
You're saying that you have a folder with MP4/MKV files that doesn't thumbnail or only thumbnails incredibly slowly, is that correct?

In most cases this means that one or more of these files does not currently decode correctly and may be corrupt. 
If there is a specific file where the thumbnailing completely stops, when it reaches that file, please send it to me, and I'll investigate.
Alternatively, you could try and run these files through [MKVToolNix](https://mkvtoolnix.download/downloads.html#windows), and see if that fixes the thumbnailing with Icaros.

**Reply by Xanashi** [reply]
@showmeren I will try to reproduce this issue, once I get my new drive installed.

**Reply by chunme** [reply]
真的感谢您认真的回复

我明白了你说的“备份”缓存，是加速缩略图的加载，我也试过很多次，
甚至将缓存的文件 Icaros_idx.icdb / Icaros_1280.icdb 修改为 thumbcache_idx.db / thumbcache_1280.db
系统仍然无法调用它来完成缩略图的显示。还是和原来一样加载超慢 最后卡主，以导致  资源管理器占用 CPU 100%

使用 MediaPreview生成的缓存文件就可以显示缩略图（如图），部分不显示的安装 K-Lite Codec Pack 勾选 IcarosThumbnailProvider 即可显示
![Snipaste_2023-12-14_03-06-41](https://github.com/Xanashi/Icaros/assets/146305362/a1f72c21-27bb-412a-b26f-0c6d3af1b184)
![Snipaste_2023-12-14_03-38-41](https://github.com/Xanashi/Icaros/assets/146305362/c15ebd4f-0991-49c7-9cbb-6aa5f7edd5c7)

**我在想如果 Icaros 生成的不是 Icaros_1280.icdb 而是 thumbcache_1280.db 格式的缓存文件或许可以解决我遇到的问题。**

**Comment by NTassArt** [comment]
I have a question about the overlay icons and if anyone else has had this issue. I am running ver. 3.3.2, first time user (was using MystictThumbs before) . I installed a Icaros a week or 2 after I updated Adobe CC to 2024 version and the Adobe apps show a broken overlay icon for Photoshop and Illustrator but Xd icon overlay shows up no problem. I'm not sure about other Adobe apps as I don't have any saved currently to compare.

Please let me know if there is a fix to this issue as its a little annoying for me to see those broken icon overlays.

![Screenshot 2024-03-03 193026](https://github.com/Xanashi/Icaros/assets/162068527/372ac4be-0e05-49cf-affe-4973a342e40a)
![Screenshot 2024-03-03 193055](https://github.com/Xanashi/Icaros/assets/162068527/3dc9193e-5cf2-4813-83da-5b8986a7781d)
![Screenshot 2024-03-03 193117](https://github.com/Xanashi/Icaros/assets/162068527/fe6e0c6f-c9ef-4dea-ac24-7bf1aa7a684c)

**Reply by clsid2** [reply]
Explorer is responsible for those icon overlays. Icaros is not involved.

Google how to "clear windows icon cache" and then reboot.

**Reply by NTassArt** [reply]
Thanks for the prompt reply clsid2.

I have done the clear icon and thumbnail cache 3 times already with no luck. I was hoping you'd have a solution to this problem.

**Reply by NTassArt** [reply]
Hello again,

I wanted to update you on what I have done since the last comment.

I have uninstalled Adobe CC 2024 and Icaros.

I have cleared the icon and thumbnail cache.

I have removed all residual files and folders left behind by Adobe and Icaros.

Ran a registry cleaner and manually removed any registry entries that were not removed automatically by the registry cleaner.

Restarted the computer and performed the registry clean and icon/thumbnail delete again with another restart after.

Installed Adobe CC 2024 again and restart the computer, then installed Icaros.

After install of Icaros I allocated the .psd, .psdt, .ai, .xd, .eps file extensions I frequently use and activated the thumbnail previews with properties and cache enabled (dynamic). 

Icaros prompted me that the .PSD file extension was in use my another application. PSD and AI files still show broken icon overlay as in the image above.

I installed Icaros on a whim to an older laptop running Windows 7 with Adobe CS4 and I got the AI extension files to show the AI icon overlay but the PSD file extension was now a blank icon overlay like the image above for the PSDT file extension I attached the other day.

Any idea why this is happening on Windows 11 and why I got the message ".psd file extension in use by another application"?

Thanks in advance,
N Tass Art

**Reply by clsid2** [reply]
It must be related to how you are creating file associations for those extensions.

https://www.winhelponline.com/blog/change-default-icon-file-type-windows/

**Reply by NTassArt** [reply]
Hello again clsid2,

I have checked all the file extension associations as well using the above tool you linked as well as another app "Type" and even checked the registry after install to see if everything was associated correctly. All are associated correctly yet I'm now getting AI files icon overlay but the PSD and XD icon overlay is blank missing.

I am also still getting the message that the PSD files extension is being used by another app, which I haven't had any luck finding out which app it is. I even downloaded stand alone icons for the Adobe apps and changed the associations to the designated icon files with out any luck either. So I'm stumped as to how this issue is going to get resolved.

Thank you for all you advice and assistance so far.
N Tass Art

![Screenshot 2024-03-05 114121](https://github.com/Xanashi/Icaros/assets/162068527/fdfcf610-6437-405f-accb-f298b4301cdf)
![Screenshot 2024-03-05 114145](https://github.com/Xanashi/Icaros/assets/162068527/bff2a6b7-0acb-4aed-82e2-1f3cd5c04127)
